Client/Frontend/Browser/Toolbars/BottomToolbar/Menu/MenuViewController.swift

```diff
@@ -1,405 +1,287 @@
-            guard let menuCell = cell as? MenuCell else { return }
-            openVPNAction(menuCell: menuCell)
-        case .bookmarks: openBookmarks()
-        case .history: openHistory()
-        case .settings: openSettings()
-        case .add: openAddBookmark()
-        case .share: openShareSheet()
-        case .downloads: openDownloads()
+    override func dismiss(animated flag: Bool, completion: (() -> Void)? = nil) {
+        if let _ = presentedViewController as? InnerMenuNavigationController,
+           presentingViewController?.presentedViewController === self {
+            isDismissing = true
+            presentingViewController?.dismiss(animated: flag, completion: completion)
+        } else {
+            super.dismiss(animated: flag, completion: completion)
         }
     }
     
-    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return visibleButtons.count
+    private var isPresentingInnerMenu: Bool {
+        if let _ = presentedViewController as? InnerMenuNavigationController {
+            return true
+        }
+        return false
     }
-    
-    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let button = visibleButtons[indexPath.row]
-        let cell = { () -> MenuCell in
-            guard let button = MenuButtons(rawValue: indexPath.row) else {
-                assertionFailure("No cell with \(indexPath.row) tag.")
-                return MenuCell()
+}
+
+extension MenuViewController: PanModalPresentable {
+    var panScrollable: UIScrollView? {
+        // For SwiftUI:
+        //  - in iOS 13, ScrollView will exist within a host view
+        //  - in iOS 14, it will be a direct subview
+        // For UIKit:
+        //  - UITableViewController's view is a UITableView, thus the view itself is a UIScrollView
+        //  - For our non-UITVC's, the scroll view is a usually a subview of the main view
+        func _scrollViewChild(in parentView: UIView, depth: Int = 0) -> UIScrollView? {
+            if depth > 2 { return nil }
+            if let scrollView = parentView as? UIScrollView {
+                return scrollView
             }
-            
-            switch button {
-            case .vpn:
-                let menuCell =  MenuCell(withToggle: true, fullLineSeparator: true)
-                
-                switch BraveVPN.vpnState {
-                case .notPurchased, .purchased, .expired:
-                    break
-                case .installed(let enabled):
-                    menuCell.toggleButton.isOn = enabled
-                    menuCell.isLoading = BraveVPN.reconnectPending
+            for view in parentView.subviews {
+                if let scrollView = view as? UIScrollView {
+                    return scrollView
+                }
+                if !view.subviews.isEmpty, let childScrollView = _scrollViewChild(in: view, depth: depth + 1) {
```

Comment:
I have 2 questions here,
- First one I guess we are doing this check for iOS 13 only, If thats the case can we mark it like that
- Second question is Can there be a situation where we have a scrollview as a direct subview and also scroll view within a hostview? If yes shouldnt we pick the scrollview which exists as direct subview ?

Reply:
1. Not 13 only, 14 can also have subviews as subview of the `view` when we create custom VC's
2. Yes but we always want the top-level one we find since its the one that controls whether or not panning begins to open/close the pan modal. If its the direct view it will return a few lines above, otherwise we check 1 more level deep. Any other scrolls within are assumed to function separately. I guess one improvement would be to find the first one that scrolls vertically, but Im not sure how we could go about doing that